api/_dexes/0x/allowance-holder.ts

```diff
@@ -82,13 +82,17 @@ export function get0xStrategy(
             }
           : // We need to invert the include sources to be compatible with the API
             // because 0x doesn't support the `includeSources` parameter
-            sources?.sourcesType === "include"
+            sources?.sourcesType === "include" &&
+              sources.sourcesNames?.length > 0
             ? {
                 excludedSources: SOURCES.sources[swap.chainId]
-                  .map((s) => s.key)
                   .filter(
-                    (sourceKey) => !sources.sourcesKeys.includes(sourceKey)
+                    ({ names }) =>
+                      !sources.sourcesNames?.some((sourceName) =>
```

Comment:
```suggestion
                      !sources.sourcesNames.some((sourceName) =>
```

Also, just wondering, why are we now filtering by name here and key in other strategies?

Reply:
No you are right... This should actually be keys that we compare